### PR TITLE
Add support for Basic Auth to ConsulHealthCheck

### PR DESCRIPTION
**Pull request by riker09**
I have read [Don't push your pull request](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/). I'm sorry if my PR offends any maintainer. You guys are doing great work.

But hopefully this change is so minimal, that it will not affect the greater scheme of things in any unexpected way.

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Hi @riker09 

I try to review this ASAP, thanks for contributing!

File: src/HealthChecks.Consul/ConsulOptions.cs [hunk 1]
```diff
@@ -7,5 +7,9 @@ public class ConsulOptions
         public int Port { get; set; }
 
         public bool RequireHttps { get; set; }
+
```
Comment (unaizorrilla) [inline src/HealthChecks.Consul/ConsulOptions.cs]: Could be a good idea to add RequireBasicAuthentication as boolean and use it to add the header in the request?

File: src/HealthChecks.Consul/ConsulHealthCheck.cs [hunk 2]
```diff
@@ -19,7 +20,14 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var result = await _httpClientFactory()
+                var client = _httpClientFactory();
+
+                if (!String.IsNullOrEmpty(_options.Username) && !String.IsNullOrEmpty(_options.Password)) {
```
Comment (unaizorrilla) [inline src/HealthChecks.Consul/ConsulHealthCheck.cs]: Probably check RequireBasicAuthentication would be better

**Comment by riker09** [comment]
Sounds like a good idea to enable BasicAuth explicitely. I will implement that next week.

**Comment by unaizorrilla** [comment]
Hi @riker09 

Merged, a release build is on appveyor, when finish a new nuget package 2.2.2 for Consul healthcheck will be available!

**Pull request merged**